hexrdgui/main_window.py

```diff
@@ -1730,3 +1735,55 @@ def action_apply_median_filter_toggled(self, b):
         # The dialog should have modified HexrdConfig's median filter options
         # already. Just apply it now.
         HexrdConfig().apply_median_filter_correction = b
+
+    def validateDragDropEvent(self, event):
+        mime_data = event.mimeData()
+        if not mime_data.hasUrls():
+            event.ignore()
+            return
+
+        # Make sure the selction(s) are files with a supported file extensions
+        paths = [url.toLocalFile() for url in mime_data.urls()]
+        supported_extensions = [
+            '.hexrd',
+            '.yml', '.yaml',
+            '.h5', '.hdf5',
+            '.tif', '.tiff',
+            '.png',
+            '.jpg', '.jpeg'
+        ]
+        exts = [Path(path).suffix.lower() for path in paths]
+        if any(ext not in supported_extensions for ext in exts):
+            event.ignore()
+            return
+
+        if event.type() == QEvent.Drop:
+            self.dropEvent(paths)
+        else:
+            event.acceptProposedAction()
+
+    def dropEvent(self, paths):
+        ext = Path(paths[0]).suffix.lower()
+        if len(paths) == 1:
+            try:
+                # if hdf5 try as state, then materials, then images
+                if ext in ('.h5', '.hdf5'):
+                    try:
+                        # Try loading it as a state file first
+                        self.load_state_file(paths[0])
+                    except:
+                        # If that fails, try loading it as a materials file
+                        HexrdConfig().load_materials(paths[0])
```

Comment:
I think it's true that we normally raise exceptions when we attempt to load a file that isn't the right kind. Since we are now relying on it, now we should definitely make sure we do that :slightly_smiling_face:.

Reply:
I've made some improvements here and we do get an error when trying to load non-image file(s):
```
Failed to process imageseries.

zero length imageseries.
```

If you try to load an invalid image file you will still have to go through the steps of selecting the path (for `.h5`, `.hdf5`) and associating the image(s) with the correct detectors before you get the error, both of which can be cancelled if you realize this was a mistake.